cursive/src/main.rs

```diff
@@ -63,10 +63,20 @@ fn copy(ui: &mut Cursive) -> () {
 }
 
 fn open(ui: &mut Cursive) -> () {
-    let password_entry: pass::PasswordEntry = (*ui
+    let password_entry_option: Option<Option<std::rc::Rc<ripasso::pass::PasswordEntry>>> = ui
         .call_on_id("results", |l: &mut SelectView<pass::PasswordEntry>| {
-            l.selection().unwrap()
-        }).unwrap()).clone();
+            l.selection()
+        });
+
+    let password_entry: pass::PasswordEntry = (*(match password_entry_option {
```

Comment:
This could be expressed more cleanly using ? instead of pattern matching

Reply:
I can't use ? due to that cursive requires the method signature to be () i think.

I can rework it using Option::is_none() like this:

```
-    let password_entry: pass::PasswordEntry = (*(match password_entry_option {
-        Some(level_1) => {
-            match level_1 {
-                Some(level_2) => level_2,
-                None => return
-            }
-        },
-        None => return
-    })).clone();
+    if password_entry_option.is_none() {
+        return;
+    }
+    let password_entry_option_unwrapped = password_entry_option.unwrap();
+    if password_entry_option_unwrapped.is_none() {
+        return;
+    }
+
+    let password_entry: pass::PasswordEntry = (*(password_entry_option_unwrapped.unwrap())).clone();
```

But i'm not very sure that it's an improvement.